app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -310,7 +310,7 @@ def preallocate_and_clone_disks(args, template)
           {:id => volume[:id], :sparse => 'false', :format => 'raw', :storage_domain => volume[:storage_domain]}
         else
           template_volume = template_disks.detect {|v| v.id == volume["id"]}
-          {:id => volume["id"], :storage_domain => volume["storage_domain"]} if template_volume.storage_domain != volume["storage_domain"]
+          {:id => volume["id"], :storage_domain => volume["storage_domain"]} if template_volume&.storage_domain != volume["storage_domain"]
```

Comment:
if `template_volume` is `nil` (i.e. no template disks match the volume id), and `volume["storage_domain"]` isn't, won't this update the wrong volume wit the storage domain?

Reply:
If `template_volume` is nil there is no disk coming from the template that is relevant here to clone and change so it shouldn't update anything.

This is a case that only got here because of another bug:
When you select an image, save and there is an error the template is changed to "Blank template" but the disk is still there with the id of the template disk.
We shouldn't clear the template or we should clear the disk like we do when you reselect the template.

I still wanted to protect the access to `storage_domain` because detect can theoretically return `nil` in more cases here but I'm fine with repurposing the bug to solve the UI issue instead of opening a new bug and leave this without a nil check.